plasma_framework/python_tests/tests/contracts/root_chain/test_challenge_in_flight_exit_not_canonical.py

```diff
@@ -2,22 +2,56 @@
 from eth_tester.exceptions import TransactionFailed
 
 from plasma_core.constants import NULL_ADDRESS
-
-
-def test_challenge_in_flight_exit_not_canonical_should_succeed(testlang):
-    owner_1, owner_2, amount = testlang.accounts[0], testlang.accounts[1], 100
-    deposit_id = testlang.deposit(owner_1, amount)
-    spend_id = testlang.spend_utxo([deposit_id], [owner_1], [(owner_2.address, NULL_ADDRESS, 100)])
-    double_spend_id = testlang.spend_utxo([deposit_id], [owner_1], [(owner_1.address, NULL_ADDRESS, 100)],
-                                          force_invalid=True)
-
-    testlang.start_in_flight_exit(spend_id)
-
-    testlang.challenge_in_flight_exit_not_canonical(spend_id, double_spend_id, account=owner_2)
-
-    in_flight_exit = testlang.get_in_flight_exit(spend_id)
-    assert in_flight_exit.bond_owner == owner_2.address
-    assert in_flight_exit.oldest_competitor == double_spend_id
+from plasma_core.utils.transactions import decode_utxo_id, encode_utxo_id
+from tests_utils.constants import PAYMENT_TX_MAX_INPUT_SIZE, PAYMENT_TX_MAX_OUTPUT_SIZE
+
+
+@pytest.mark.parametrize(
+    "double_spend_output_index,challenge_input_index",
+    [(i, j) for i in range(PAYMENT_TX_MAX_OUTPUT_SIZE) for j in range(PAYMENT_TX_MAX_INPUT_SIZE)]
+)
+def test_challenge_in_flight_exit_not_canonical_should_succeed(testlang, double_spend_output_index, challenge_input_index):
+    alice, bob, carol = testlang.accounts[0], testlang.accounts[1], testlang.accounts[2]
+    deposit_amount = 100
+    deposit_id = testlang.deposit(alice, deposit_amount)
+
+    tx_output_amount = deposit_amount // PAYMENT_TX_MAX_OUTPUT_SIZE
+    outputs = [(alice.address, NULL_ADDRESS, tx_output_amount)] * PAYMENT_TX_MAX_OUTPUT_SIZE
+
+    input_tx_id = testlang.spend_utxo([deposit_id], [alice], outputs=outputs)
+    blknum, tx_index, _ = decode_utxo_id(input_tx_id)
+    double_spend_utxo = encode_utxo_id(blknum, tx_index, double_spend_output_index)
+
+    ife_output_amount = tx_output_amount
+    ife_tx_id = testlang.spend_utxo(
+        [double_spend_utxo],
```

Comment:
I'd say that there still is one dimension to test out here - various indices of the double spent input in the `ife_tx` - for now you're always at `0`. I know that I might be sounding paranoid. At least test it in a separate test, if you're concerned about test running time.

Same comment for the `respond to non-canonical challenge` version of this test

Reply:
Can you take a look on `_indexes_for_test()`. Now I loops through the 3 indexes with this function which loop through each dimension with 2 other indexes being fixed.